test/custom_resource_crud/custom_resource_crud_test.go

```diff
@@ -73,10 +74,205 @@ var _ = Describe("crcrud test", func() {
+			indexerPodName = fmt.Sprintf(testenv.IndexerPod, deployment.GetName(), 2)
+			testenv.VerifyCPULimits(deployment, testenvInstance.GetName(), indexerPodName, defaultCPULimits)
+
+			// Change CPU limits to trigger CR update
+			idxc := &enterprisev1.IndexerCluster{}
+			instanceName := fmt.Sprintf("%s-idxc", deployment.GetName())
+			err = deployment.GetInstance(instanceName, idxc)
+			if err != nil {
+				testenvInstance.Log.Error(err, "Unable to fetch Indexer Cluster deployment")
+			}
+			idxc.Spec.Resources.Limits = corev1.ResourceList{
+				"cpu": resource.MustParse(newCPULimits),
+			}
+			err = deployment.UpdateCR(idxc)
+			Expect(err).To(Succeed(), "Unable to deploy Indexer Cluster with updated CR")
+
+			// Verify Indexer Cluster is updating
+			idxcName := deployment.GetName() + "-idxc"
+			testenv.VerifyIndexerClusterPhase(deployment, testenvInstance, splcommon.PhaseUpdating, idxcName)
+
+			// Verify Indexers go to ready state
+			testenv.VerifyIndexerClusterPhase(deployment, testenvInstance, splcommon.PhaseReady, idxcName)
+
+			// Verify CPU limits after updating the CR
+			indexerPodName = fmt.Sprintf(testenv.IndexerPod, deployment.GetName(), 0)
+			testenv.VerifyCPULimits(deployment, testenvInstance.GetName(), indexerPodName, newCPULimits)
+			indexerPodName = fmt.Sprintf(testenv.IndexerPod, deployment.GetName(), 1)
+			testenv.VerifyCPULimits(deployment, testenvInstance.GetName(), indexerPodName, newCPULimits)
+			indexerPodName = fmt.Sprintf(testenv.IndexerPod, deployment.GetName(), 2)
+			testenv.VerifyCPULimits(deployment, testenvInstance.GetName(), indexerPodName, newCPULimits)
+
+			// Verify CPU limits on Search Heads before updating the CR
+			SearchHeadPodName := fmt.Sprintf(testenv.SearchHeadPod, deployment.GetName(), 0)
+			testenv.VerifyCPULimits(deployment, testenvInstance.GetName(), SearchHeadPodName, defaultCPULimits)
+			SearchHeadPodName = fmt.Sprintf(testenv.SearchHeadPod, deployment.GetName(), 1)
+			testenv.VerifyCPULimits(deployment, testenvInstance.GetName(), SearchHeadPodName, defaultCPULimits)
+			SearchHeadPodName = fmt.Sprintf(testenv.SearchHeadPod, deployment.GetName(), 2)
+			testenv.VerifyCPULimits(deployment, testenvInstance.GetName(), SearchHeadPodName, defaultCPULimits)
+
+			// Change CPU limits to trigger CR update
+			shc := &enterprisev1.SearchHeadCluster{}
+			instanceName = fmt.Sprintf("%s-shc", deployment.GetName())
+			err = deployment.GetInstance(instanceName, shc)
+			if err != nil {
+				testenvInstance.Log.Error(err, "Unable to fetch Search Head Cluster deployment")
+			}
+
+			shc.Spec.Resources.Limits = corev1.ResourceList{
+				"cpu": resource.MustParse(newCPULimits),
+			}
+			err = deployment.UpdateCR(shc)
+			Expect(err).To(Succeed(), "Unable to deploy Search Head Cluster with updated CR")
+
+			// Verify Search Head Cluster is updating
+			testenv.VerifySearchHeadClusterPhase(deployment, testenvInstance, splcommon.PhaseUpdating)
```

Comment:
Move Line 142 - 165 above 106 before updating Indexer Cluster CR. This will reduce test run time as SHC take the longest time to apply updates.